z88dk/z88dk
# ticks: complex compressions evaluation with “print” (or “p”)

This change adds implementation to the `print` command of ticks, which was unimplemented prior to this.
It uses flex/bison to parse expression according to `expressions.l`/`expressions.y` and with help of `exp_engine.c` becomes able to do C stuff like referencing, de-referencing, pointer math, type cast, etc.

**How to test it**
- Pull this example project: https://github.com/desertkun/z88dk-debug-error-minimum-example
- compile it with make
- run `z88dk-ticks -x example.map -d -l 0x8000 -pc 0x8000 example__.bin`
- but a breakpoint over method "check" `b check`
- continue `c`
- `info locals`
- `p kek`
- `p test`
- `p (uint8_t)1`
- `p *kek + 10`
- `p (uint8_t*)test`
- etc

**Limitations**
- Structs are not yet supported, only primitive types
- No pointers of pointers
- ~~Only local variables can be printed out as of now, but that's not a limitation of evaluation but rather inability to obtain global variables at all~~
